torch/_decomp/decompositions.py

```diff
@@ -739,11 +739,11 @@ def slice_forward(
     elif guard_size_oblivious(start_val > sizes[dim]):
         start_val = sizes[dim]
 
-    if guard_size_oblivious(end_val < start_val):
+    if statically_known_true(end_val == sys.maxsize):
```

Comment:
i think the idea here, if end was passed as None then it become sys.maxsize. 
and start was unbacked we do not want end_val = start_val but rather we want, end_val = sizes[dim].